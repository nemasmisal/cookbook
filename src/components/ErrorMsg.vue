<template>
  <div class="error" v-for="error of errorsObj" :key="error.$message">
    {{ error.$message }}
  </div>
</template>

<script>
export default {
  name: 'InputErrMsgTemp',
  props: {
    errorsObj: {
      type: Object,
      required: true,
    },
  },
};
</script>

<style></style>
