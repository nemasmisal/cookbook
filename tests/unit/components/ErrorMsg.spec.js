import { shallowMount } from '@vue/test-utils';
import ErrorMsg from '@/components/ErrorMsg.vue';

describe('ErrorMsg component tests', () => {
  let errorsObj = [];
  let wrapper;
  beforeEach(() => {
    errorsObj = [
      { $message: 'first error msg' },
      { $message: 'second error msg' },
    ];
    wrapper = shallowMount(ErrorMsg, {
      props: { errorsObj },
    });
  });
  describe('Testing component props', () => {
    it('Should accept/use passed props', () => {
      expect(wrapper.vm.errorsObj).toMatchObject(errorsObj);
    });
  });
  describe('Testing component rendering', () => {
    it('Should have expected HTML structure', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
