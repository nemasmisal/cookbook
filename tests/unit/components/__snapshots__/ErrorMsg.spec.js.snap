// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMsg component tests Testing component rendering Should have expected HTML structure 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="error"
  >
    first error msg
  </div>
  <div
    class="error"
  >
    second error msg
  </div>
  
</div>
`;
